Zend/zend_alloc.c

```diff
@@ -3070,25 +3106,28 @@ static void alloc_globals_ctor(zend_alloc_globals *alloc_globals)
 	tmp = getenv("USE_ZEND_ALLOC");
 	if (tmp && !ZEND_ATOL(tmp)) {
 		bool tracked = (tmp = getenv("USE_TRACKED_ALLOC")) && ZEND_ATOL(tmp);
-		zend_mm_heap *mm_heap = alloc_globals->mm_heap = malloc(sizeof(zend_mm_heap));
+		ZEND_ASSERT(sizeof(zend_mm_heap) <= zend_mm_get_page_size());
+		zend_mm_heap *mm_heap = alloc_globals->mm_heap = malloc(zend_mm_get_page_size() * 2);
 		memset(mm_heap, 0, sizeof(zend_mm_heap));
 		mm_heap->use_custom_heap = ZEND_MM_CUSTOM_HEAP_STD;
 		mm_heap->limit = (size_t)Z_L(-1) >> 1;
 		mm_heap->overflow = 0;
 
+		mprotect(ZEND_MM_RO_HEAP(mm_heap), zend_mm_get_page_size(), PROT_WRITE);
```

Comment:
In this case we have no guarantee that `mm_heap` is page-aligned, so this call may fail. I think we can use `zend_mm_chunk_alloc_int()` to allocate `->mm_heap` above (and `zend_mm_munmap()` to free it).

Reply:
It seems that I messed up something wrt. zend_mm_chunk_alloc_int/zend_mm_munmap, but I can't find what :/

Reply:
I will check if I can find something